db/pagination/pagination.go

```diff
@@ -37,12 +37,12 @@ const (
 type Paginator struct {
 	Key    string
 	Order  string
-	Limit  uint64
+	Limit  int64
 	Offset uint64
 }
 
 //NewPaginator create Paginator
-func NewPaginator(s *schema.Schema, key, order string, limit, offset uint64) (*Paginator, error) {
+func NewPaginator(s *schema.Schema, key, order string, limit int64, offset uint64) (*Paginator, error) {
```

Comment:
most of these parameters are in fact optional. could we refactor this code a bit?